aspnetcore/tutorials/choose-web-ui.md

```diff
@@ -0,0 +1,195 @@
+---
+title: Choose an ASP.NET Core UI
+author: wadepickett
+description: Understand when to use the various ASP.NET Core web UI technologies Microsoft provides and supports.
+ms.author: wpickett
+ms.date: 06/30/2021
+no-loc: [Home, Privacy, Kestrel, appsettings.json, "ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]
+uid: tutorials/choose-web-ui
+---
+# Choose an ASP.NET Core web UI
+
+ASP.NET Core is a complete end to end UI framework.  You choose which functionalities to combine that fit your web UI app needs.
+
+Start by determining if your web UI app will primarily benefit from pre-rendering UI on the server, or rendering UI at the client browser. Then review the server rendered or client rendered ASP.NET Core project types to find which best fit your apps needs. 
+
+## Choose an ASP.NET Core web UI technology
+
+Use the following flowchart to select an ASP.NET Core web UI technology:
+
+![Decision tree for ASP.NET Core web UI technology](choose-web-ui/_static/choose_ui.png)
+
+## Choose server rendered UI or client rendered UI
```

Comment:
This really isn't an up front one time choice for a given app. You often end up having both server-side and client-side UI rendering in the same web app. For example, in an MVC app most of the rendering happens from the server, but you still end up using client-side logic for form validation. In Blazor apps you prerender UI from the server, but then most of the UI interactions get handled client-side. Instead of framing this as an up front decision, I think we'll need to structure this doc more around common scenarios/patterns and the ASP.NET Core technologies that you use to implement them.

Reply:
I changed this from a hard decision structure to more informative of Server side and Client side benefits awareness to help shape the decision, and then followed by benefits of each app architecture (project type).  Understanding which ones "primarily" do the work on the server or client still seemed helpful.  I also repositioned Blazor to focus on Blazor components first, then the advantages of the hosting options.